articles/node_lifecycle.md

```diff
@@ -0,0 +1,287 @@
+
+A node must be deactivated before it can be destroyed.
+
+On entering this state, the node's `onDestroyed` method must be run. This is
+the node's last chance to clean up.
+
+Implementation note: In object-oriented languages, the onDestroyed method can
+be mapped to the class destructor.
+
+
+
+
+## Interface
+
+A node must provide the messages and services defined in this section to
+qualify as a managed node. These services define the external interface of a
+managed node as seen by tools that perform the managing. As such, it should be
+possible to consider any object that provides these services correctly to be a
+managed node.
+
+Conversely, any object that provides these services but does not behave in the
+way defined in the life cycle state machine is malformed.
+
+These services may be provided via attributes and method calls (for local
+management) or via ROS messages and services (for remote management).
+In the case of providing a ROS middleware interface, specific topics must be
+used, and they should be placed in a suitable namespace.
+
+### Initialisation
+
+The following service should be called to initialise the node. This service must
+be named `initialise`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Activation
+
+The following service should be called to activate the node. This service must
+be named `activate`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Deactivation
+
+The following service should be called to deactivate the node. This service
+must be named `deactivate`.
+
+{% raw %}
+    ---
+    bool success
+    string result_msg
+{% endraw %}
+
+
+### Clearing the error state
+
+The following service should be called to clear the node's fatal error status.
+This is done when the node is to be re-activated in the hope that the error has
+gone away.
+
+{% raw %}
+    ---
+    bool success
+{% endraw %}
+
+### State reporting
+
+The following service should be called to query the current state of the node.
+The service must be named `get_state`.
+
+{% raw %}
+    ---
+    int8 created=0
+    int8 inactive=1
+    int8 active=2
+    int8 error=3
+    int8 destroyed=4
+
+    int8 state
+{% endraw %}
+
+The value of `state` must be one of the constants defined in the service
+definition.
+
+
+## Composite nodes
+
+A composite node is a system-level concept of one node being made up of one or
+more other nodes as its implementation, rather than direct implementation using
+source code.
+
+It is conceivable that all child nodes in a composite node will be registered
+to a single Executor that is managed by the composite node. Such a design
+allows for grouping nodes into a single process, such as for control loops.
+(See the Executor examples on GitHub for an example of this.)
+
+A composite node should enforce its own life cycle state on all its children
+according to the following mapping:
```

Comment:
The composite node can enforce triggers of its own lifecycle on its children, but nothing prevents its children from diverging from the parent lifecycle, e.g., they could deactivate themselves or each other while the parent is active.

The table below could be complemented with another one specifying what action the parent takes when its children step though their lifecycle (without parent intervention). You already mention what should happen when a child enters `fatal_error`. 

More below...